smrf/distribute/wind.py

```diff
@@ -208,61 +208,67 @@ def initialize(self, topo, data):
-                uv[:,1]=self.Y.flatten()
-                uv[:,0]=self.X.flatten()
-
-                self.vtx, self.wts = utils.interp_weights(xy, uv,d=2)
-                ###### end do this first to speedup the interpolation later ####
+                self.initialize_wind_ninja(topo)
 
         if not self.distribute_drifts:
             # we have to pass these to precip, so make them none if we won't use them
             self.dir_round_cell = None
             self.cellmaxus = None
 
+    def initialize_wind_ninja(self, topo):
+
+        # WindNinja output height in meters
+        self.wind_height = float(self.config['wind_ninja_height'])
+        # set roughness that was used in WindNinja simulation
+        # WindNinja uses 0.01m for grass, 0.43 for shrubs, and 1.0 for forest
+        self.wn_roughness = float(self.config['wind_ninja_roughness']) * \
+                            np.ones_like(topo.dem)
+
+        # get our effective veg surface roughness
+        # to use in log law scaling of WindNinja data
+        # using the relationship in
+        # https://www.jstage.jst.go.jp/article/jmsj1965/53/1/53_1_96/_pdf
+        self.veg_roughness = topo.veg_height / 7.39
+        # make sure roughness stays reasonable using bounds from
+        # http://www.iawe.org/Proceedings/11ACWE/11ACWE-Cataldo3.pdf
+
+        self.veg_roughness[self.veg_roughness < 0.01] = 0.01
+        self.veg_roughness[np.isnan(self.veg_roughness)] = 0.01
+        self.veg_roughness[self.veg_roughness > 1.6] = 1.6
+
+        # precalculate scale arrays so we don't do it every timestep
+        self.ln_wind_scale = np.log((self.veg_roughness + self.wind_height) / self.veg_roughness) / \
+                                np.log((self.wn_roughness + self.wind_height) / self.wn_roughness)
+
+        ###### do this first to speedup the interpolation later ####
+        # find vertices and weights to speedup interpolation fro ascii file
+        fmt_d = '%Y%m%d'
+        fp_vel = glob.glob(os.path.join(self.wind_ninja_dir,
+                                'data{}'.format(self.start_date.strftime(fmt_d)),
+                                'wind_ninja_data',
+                                '*_vel.asc'))[0]
+
+        # get wind ninja topo stats
+        ts2 = utils.get_asc_stats(fp_vel)
+        self.windninja_x = ts2['x'][:]
+        self.windninja_y = ts2['y'][:]
+
+        XW, YW = np.meshgrid(self.windninja_x, self.windninja_y)
+        xwint = XW.flatten()
+        ywint = YW.flatten()
+        self.wn_mx = xwint
+        self.wn_my = ywint
+
+        xy = np.zeros([XW.shape[0]*XW.shape[1],2])
+        xy[:,1] = ywint
+        xy[:,0] = xwint
+        uv = np.zeros([self.X.shape[0]*self.X.shape[1],2])
+        uv[:,1] = self.Y.flatten()
+        uv[:,0] = self.X.flatten()
+
+        self.vtx, self.wts = utils.interp_weights(xy, uv,d=2)
+        ###### end do this first to speedup the interpolation later ####
+
```

Comment:
So you moved the windninja interpolation into its own function. Nice. I like.

Reply:
This is just the initialization of wind ninja, it used to be in the `initialize` function but this makes it easier to test

Reply:
Ah. I see. So it just gets the coordinates and the interpolation weights.